### PR TITLE
[DROOLS-6880] make dtables to work only with .drl.xls extension

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6880

Related PRs:
https://github.com/kiegroup/drools/pull/4278
https://github.com/kiegroup/kogito-runtimes/pull/2105
https://github.com/kiegroup/kogito-examples/pull/1196

### CODE REVIEW

**Comment by danielezonca** [comment]
Please add a note in Kogito release notes

File: drools-drl-quarkus-extension/drools-drl-quarkus-deployment/src/main/java/org/drools/drl/quarkus/deployment/DroolsCompilationProvider.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 public class DroolsCompilationProvider extends AbstractCompilationProvider {
 
-    private static final Set<String> MANAGED_EXTENSIONS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(".drl", ".xls", ".xlsx", ".csv")));
+    private static final Set<String> MANAGED_EXTENSIONS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(".drl", ".drl.xls", ".drl.xlsx", ".drl.csv")));
```
Comment (danielezonca) [inline drools-drl-quarkus-extension/drools-drl-quarkus-deployment/src/main/java/org/drools/drl/quarkus/deployment/DroolsCompilationProvider.java]: What about centralise and use ResourceType also here?
```java
Set<String> MANAGED_EXTENSIONS = new HashSet<>();
MANAGED_EXTENSIONS.addAll(ResourceType.DRL.getAllExtensions());
MANAGED_EXTENSIONS.addAll(ResourceType.DTABLE.getAllExtensions());
```

**Review by danielezonca — APPROVED** [review]
One minor comment

**Comment by kie-ci1** [comment]
**(tests) - kogito-examples job** `#402` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 451
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-examples/402/testReport/org.acme.it/QueryAnswerServiceIT/Build_projects___queryAnswerProcessCycle/">Build projects / org.acme.it.QueryAnswerServiceIT.queryAnswerProcessCycle</a></summary>
java.lang.AssertionError: <br/>1 expectation failed.<br/>Expected status code <201> but was <400>.<br/><br/>	at org.acme.it.QueryAnswerServiceIT.queryAnswerProcessCycle(QueryAnswerServiceIT.java:89)<br/>
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-examples/402/testReport/org.kie.kogito.springboot.outbox/OutboxIT/testSendProcessEvents/">org.kie.kogito.springboot.outbox.OutboxIT.testSendProcessEvents</a></summary>
org.awaitility.core.ConditionTimeoutException: <br/>Assertion condition defined as a lambda expression in org.kie.kogito.springboot.outbox.OutboxIT 1 expectation failed.<br/>Expected status code <200> but was <404>.<br/> within 1 minutes.<br/>	at org.kie.kogito.springboot.outbox.OutboxIT.testSendProcessEvents(OutboxIT.java:129)<br/>Caused by: java.lang.AssertionError: <br/>1 expectation failed.<br/>Expected status code <200> but was <404>.<br/><br/>	at org.kie.kogito.springboot.outbox.OutboxIT.lambda$testSendProcessEvents$1(OutboxIT.java:136)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-examples/402/display/redirect

File: drools-drl-quarkus-extension/drools-drl-quarkus-deployment/src/main/java/org/drools/drl/quarkus/deployment/DroolsCompilationProvider.java (hunk 1 above)
Comment (mariofusco) [reply drools-drl-quarkus-extension/drools-drl-quarkus-deployment/src/main/java/org/drools/drl/quarkus/deployment/DroolsCompilationProvider.java]: Done.

**Pull request merged**